File: bridge_iq/environment.py
"""
Environment configuration for the BridgeIQ client.

This module provides environment-specific configuration for different 
deployment scenarios (production, testing).
"""
from enum import Enum
from typing import Dict, Optional

class Environment(str, Enum):
    """Environment configuration for the BridgeIQ client."""
    
    PRODUCTION = "production"
    TESTING = "testing"
    
    @classmethod
    def from_string(cls, env_string: str) -> "Environment":
        """Create an Environment from a string value.
        
        Args:
            env_string: String representation of environment
            
        Returns:
            Environment instance
            
        Raises:
            ValueError: If the string doesn't match a valid environment
        """
        try:
            return cls(env_string.lower())
        except ValueError:
            valid_values = ", ".join([e.value for e in cls])
            raise ValueError(
                f"Invalid environment: {env_string}. "
                f"Valid values are: {valid_values}"
            ) 

File: src/environment.ts
/**
 * Environment configuration for the BridgeIQ client.
 * 
 * This module provides environment-specific configuration for different 
 * deployment scenarios (production, testing).
 */

/**
 * Environment types for the BridgeIQ client.
 */
export enum Environment {
  PRODUCTION = 'production',
  TESTING = 'testing',
}

/**
 * Create an Environment from a string value.
 * 
 * @param envString - String representation of environment
 * @returns Environment enum value
 * @throws Error if the string doesn't match a valid environment
 */
export function fromString(envString: string): Environment {
  const lowerEnv = envString.toLowerCase();
  
  if (Object.values(Environment).includes(lowerEnv as Environment)) {
    return lowerEnv as Environment;
  }
  
  const validValues = Object.values(Environment).join(', ');
  throw new Error(`Invalid environment: ${envString}. Valid values are: ${validValues}`);
} 